src/seedu/addressbook/commands/Command.java

```diff
@@ -26,6 +26,15 @@ public Command(int targetIndex) {
     protected Command() {
     }
 
+    /**
+     * @return true if the object calling the method is an instance of a mutating Command subclass
+     */
+    public boolean isMutating() {
+        return (this instanceof AddCommand ||
+                this instanceof DeleteCommand ||
+                this instanceof ClearCommand);
```

Comment:
This is not polymorphism.. Check out other PRs, e.g. https://github.com/nus-cs2103-AY1718S2/addressbook-level3/pull/601